ClemBot.Bot/bot/cogs/search_cog.py

```diff
@@ -0,0 +1,159 @@
+import logging
+import re
+import json
+import aiohttp
+import discord
+import discord.ext.commands as commands
+import bot.extensions as ext
+from enum import Enum
+from markdownify import markdownify
+from urllib.parse import urlencode
+from urllib.parse import unquote_plus
+from bot.consts import Colors
+from bot.messaging.events import Events
+
+log = logging.getLogger(__name__)
+max_related_topics = 5
+image_url = 'https://duckduckgo.com'
+search_url = 'https://api.duckduckgo.com/'
+icon_url = 'https://i.imgur.com/Knq2kVa.png'
+related_topics_pattern = re.compile('(https://duckduckgo.com/)([acdne]/)?(\\S+)', re.IGNORECASE)
+
+
+class Category(Enum):
+    ARTICLE = 'A', 'Article'
+    DISAMBIGUATION = 'D', 'Disambiguation'
+    CATEGORY = 'C', 'Category'
+    NAME = 'N', 'Name'
+    EXCLUSIVE = 'E', 'Exclusive'
+    NONE = 'None', 'None'
+
+    def code(self):
+        return self.value[0]
```

Comment:
what does this do?

Reply:
It was my 3 a.m. brain thinking "how would i represent a category from a letter" and instead of using dicts I used an enum lmao. Changed in latest commit.